File: 1.0/src/cz/mg/vulkan/PFNvkSubmitDebugUtilsMessageEXT.java
package cz.mg.vulkan;

public class PFNvkSubmitDebugUtilsMessageEXT extends VkFunctionPointer {
    public PFNvkSubmitDebugUtilsMessageEXT() {
    }

    protected PFNvkSubmitDebugUtilsMessageEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkSubmitDebugUtilsMessageEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkSubmitDebugUtilsMessageEXT(long value) {
        setValue(value);
    }

    public PFNvkSubmitDebugUtilsMessageEXT(VkInstance instance) {
        super(instance, new VkString("vkSubmitDebugUtilsMessageEXT"));
    }

    public void call(VkInstance instance, int messageSeverity, int messageTypes, VkDebugUtilsMessengerCallbackDataEXT pCallbackData){
        callNative(getValue(), instance != null ? instance.getVkAddress() : VkPointer.getNullAddressNative(), messageSeverity, messageTypes, pCallbackData != null ? pCallbackData.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long instance, int messageSeverity, int messageTypes, long pCallbackData);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkSubmitDebugUtilsMessageEXT.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkSubmitDebugUtilsMessageEXT_callNative(JNIEnv* env, jclass jc, jlong address, jlong instance, jint messageSeverity, jint messageTypes, jlong pCallbackData){
    (void)env;
    (void)jc;
    PFN_vkSubmitDebugUtilsMessageEXT f = (PFN_vkSubmitDebugUtilsMessageEXT)jniLongToFunctionPointer(address);
    f(
        *((VkInstance*)jniLongToPointer(instance)),
        (VkDebugUtilsMessageSeverityFlagBitsEXT)messageSeverity,
        (VkDebugUtilsMessageTypeFlagsEXT)messageTypes,
        ((VkDebugUtilsMessengerCallbackDataEXT*)jniLongToPointer(pCallbackData))
    );
}

